### DISCUSSION TITLE
Set initial temperature of a buried pipe

### DISCUSSION THREAD

**Discussion by JordiGarciaCespedes** [lbl-srg/modelica-buildings#3311] [Q&A]
I am trying to set the initial value of the temperature of the medium (water) in a buried pipe. Instead of leaving the parameter `Mypipe.T_start` as `Medium.T_default,` I would like to redeclare it as the corresponding ground temperature, since the pipe is coupled to the ground. For that matter, the model would need first to evaluate the undisturbed ground temperature at t=0 taking into account depth, soil properties and climatic data.
So far, I tried to define the value as follows (conceptually):
```Modelica
parameter Modelica.Units.SI.Temperature T_startatpipes;
Mypipe (T_start = T_startatpipes);
initial equation
    T_startatpipes=GroundCoupling.soi.T;
equation
[...]
```
But it returns error.
Any suggestions?
Thanks

**Comment by mwetter** [comment]
You should be able to simply set the value of `T_start_in`. Simulating
```
simulateModel(
  "Buildings.Fluid.Geothermal.BuriedPipes.Examples.SingleBuriedPipe(pip(T_start_in=283.15, show_T=true))",
  stopTime=3600,
  method="Cvode",
  tolerance=1e-06,
  resultFile="SingleBuriedPipe");
```
yields
![image](https://user-images.githubusercontent.com/1329080/228072594-229611ca-85f9-44c8-98a2-6af5de8e9c9e.png)

**Reply by JordiGarciaCespedes** [reply]
Dear Michael, thanks for your reply. I understand that you recommend to set a specific value for `T_start_in` by myself. However, what I really would like is that the initial value could be set automatically by "looking" first at the ground temperature at time=0s. If we think of real conditions, it is reasonable to assume that water inside the pipes is thermalised with the ground before "switching on" the circulating pumps.

**Reply by mwetter** [reply]
I see. To enable this, the model `Buildings.BoundaryConditions.GroundTemperature.UndisturbedSoilTemperature` would need to be changed to compute an initial temperature as a `parameter`. Currently `Buildings.BoundaryConditions.GroundTemperature.UndisturbedSoilTemperature` computes the soil temperature using the *variable* `T`, which is time dependent and hence its value cannot be bound to the parameter `T_start_in` of the pipe. This would probably best be done in the context of a buried pipe that models the soil with its heat capacity rather than steady state as is done in `Buildings.BoundaryConditions.GroundTemperature.UndisturbedSoilTemperature`. In particular for 5th generation DHC systems that may be an important effect.

In the meantime, I suggest you configure the example `Buildings.BoundaryConditions.GroundTemperature.Examples.UndisturbedSoilTemperature` for your conditions and then set the temperature as specified in the earlier response if your use case does not allow these initial transients.

**Reply by JordiGarciaCespedes** [reply]
Dear Michael,
Thanks for your explanations. It happens that I managed to set initial temperature as the ground temperature by specifying an initial equation where I have two buried pipes of `nSeg` discretized segments:
```
initial equation
      groundCoupling.ports.T=fill(groundCoupling.soi.T,2,nSeg) ;
```
But Honestly, I do not really understand why this is functioning... I pressume that somehow `Medium.Temperature` gets affected by this action. Anyway, Temperature starts from ground value and aproaches to it whenever the flow is set to zero:
![image](https://user-images.githubusercontent.com/105857531/231690817-bd1f145b-3575-4f2e-aadc-848571bab6f2.png)

My problem now is to achieve a correct temperature reading with a two-port temperature sensor when flow is set to zero, since I should set a value of `TAmb` corresponding to the value of the ground, which is a time-dependent parameter. Locally, temperature readout can be callibrated by setting `TAmb` to ground temperature manually and modifying the time constant for heat transfer:
![image](https://user-images.githubusercontent.com/105857531/231709169-fe7d53bb-b6f6-4006-888a-6d0aec5f626c.png)
But this cannot hold for an entire year.
